File: CustomViewDemo/app/src/main/java/com/cyq/customview/MainActivity.java
package com.cyq.customview;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

import com.cyq.customview.drawText.DrawTextActivity;
import com.cyq.customview.flowLayout.FlowLayoutActivity;
import com.cyq.customview.paintView.PaintViewActivity;
import com.cyq.customview.shadowLayout.ShadowLayoutActivity;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {
    private Button mBtnShadowLayout;
    private Button mBtnDrawText;
    private Button mBtnPaintView;
    private Button mBtnFlowLayout;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initView();
    }

    private void initView() {
        mBtnShadowLayout = findViewById(R.id.btn_shadow_layout);
        mBtnShadowLayout.setOnClickListener(this);
        mBtnDrawText = findViewById(R.id.btn_draw_text);
        mBtnDrawText.setOnClickListener(this);
        mBtnPaintView = (Button) findViewById(R.id.btn_paint_view);
        mBtnPaintView.setOnClickListener(this);
        mBtnFlowLayout = (Button) findViewById(R.id.btn_flow_layout);
        mBtnFlowLayout.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        Intent intent = new Intent();
        switch (v.getId()) {
            case R.id.btn_shadow_layout:
                intent.setClass(this, ShadowLayoutActivity.class);
                startActivity(intent);
                break;
            case R.id.btn_draw_text:
                intent.setClass(this, DrawTextActivity.class);
                startActivity(intent);
                break;
            case R.id.btn_paint_view:
                intent.setClass(this, PaintViewActivity.class);
                startActivity(intent);
                break;
            case R.id.btn_flow_layout:
                intent.setClass(this, FlowLayoutActivity.class);
                startActivity(intent);
                break;
            default:
                break;
        }
    }
}

File: CustomViewDemo/app/src/main/java/com/cyq/customview/MainActivity.kt
package com.cyq.customview

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.cyq.customview.blur.BlurActivity
import com.cyq.customview.drawText.DrawTextActivity
import com.cyq.customview.flowLayout.FlowLayoutActivity
import com.cyq.customview.intercept.InterceptActivity
import com.cyq.customview.nineLayout.NineImageLayoutActivity
import com.cyq.customview.paintView.PaintViewActivity
import com.cyq.customview.shadowLayout.ShadowLayoutActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(), View.OnClickListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initView()
    }

    private fun initView() {
        mBtnShadowLayout.setOnClickListener(this)
        mBtnDrawText.setOnClickListener(this)
        mBtnPaintView.setOnClickListener(this)
        mBtnFlowLayout.setOnClickListener(this)
        mBtnNineLayout.setOnClickListener(this)
        mBtnBlue.setOnClickListener(this)
        mBtnIntercept.setOnClickListener(this)
    }

    override fun onClick(v: View) {
        val intent = Intent()
        when (v.id) {
            R.id.mBtnShadowLayout -> {
                intent.setClass(this, ShadowLayoutActivity::class.java)
                startActivity(intent)
            }
            R.id.mBtnDrawText -> {
                intent.setClass(this, DrawTextActivity::class.java)
                startActivity(intent)
            }
            R.id.mBtnPaintView -> {
                intent.setClass(this, PaintViewActivity::class.java)
                startActivity(intent)
            }
            R.id.mBtnFlowLayout -> {
                intent.setClass(this, FlowLayoutActivity::class.java)
                startActivity(intent)
            }
            R.id.mBtnNineLayout -> {
                intent.setClass(this, NineImageLayoutActivity::class.java)
                startActivity(intent)
            }
            R.id.mBtnBlue -> {
                intent.setClass(this, BlurActivity::class.java)
                startActivity(intent)
            }
            R.id.mBtnIntercept -> {
                intent.setClass(this, InterceptActivity::class.java)
                startActivity(intent)
            }
            else -> {
            }
        }
    }
}